src/join.rs

```diff
@@ -82,21 +91,27 @@ fn reconfigure_core(
     config_json: &ConfigMap,
     schema: &OsConfigSchema,
     remote_config: &RemoteConfiguration,
-    has_config_changes: bool,
+    has_service_config_changes: bool,
     joining: bool,
+    migrated_config_json: Option<ConfigMap>,
 ) -> Result<()> {
     if joining {
         write_config_json(&args.config_json_path, config_json)?;
     }
 
-    if has_config_changes {
+    if migrated_config_json.is_some() {
+        info!("Migrating config.json...");
+        write_config_json(&args.config_json_path, &migrated_config_json.unwrap())?;
+    }
```

Comment:
`if let` would be more idiomatic here. Generally when `unwrap` is used there usually is a more idiomatic alternative. `unwrap` in tests is perfectly fine though.

```rust
    if let Some(map) = migrated_config_json {
        info!("Migrating config.json...");
        write_config_json(&args.config_json_path, &map)?;
    }
```

Reply:
Good catch!